src/sagemaker/multidatamodel.py

```diff
@@ -0,0 +1,211 @@
+        self._create_s3_model_data_path()
+
+    @property
+    def model_data_prefix(self):
+        """Placeholder docstring"""
+        return self._model_data_prefix
+
+    @model_data_prefix.setter
+    def model_data_prefix(self, model_data_prefix):
+        """
+        Args:
+            model_data_prefix:
+        """
+        # Validate path
+        if not (model_data_prefix.startswith("s3://") and model_data_prefix.endswith("/")):
+            raise ValueError(
+                'Expecting S3 model prefix beginning with "s3://" '
+                'and ending in "/". Received: "{}"'.format(model_data_prefix)
+            )
+        self._model_data_prefix = model_data_prefix
+
+    def _create_s3_model_data_path(self):
+        """
+        Create the S3 prefix path to ensure model deployment succeeds.
+        If this path does not exist, calls to CreateModel API fails
+        """
+        bucket, model_data_path = self._parse_s3_uri(self.model_data_prefix)
+        self.s3_client.put_object(Bucket=bucket, Key=os.path.join(model_data_path, "/"))
+
+    def prepare_container_def(self, instance_type, accelerator_type=None):
+        """Return a container definition set with MultiModel mode and
+        model data and all other parameters from the original model pass
+        to the MultiDataModel constructor
+
+        Subclasses can override this to provide custom container definitions
+        for deployment to a specific instance type. Called by ``deploy()``.
+
+        Returns:
+            dict[str, str]: A complete container definition object usable with the CreateModel API
+        """
+        # Copy the trained model's image and environment variables if they exist.
+        # Models trained with FrameworkEstimator set framework specific environment variables
+        # which need to be copied over
+        if self.model:
+            container_definition = self.model.prepare_container_def(instance_type, accelerator_type)
+            self.image = self.image or container_definition["Image"]
+            self.env.update(container_definition["Environment"])
+        return sagemaker.container_def(
+            self.image,
+            env=self.env,
+            model_data_url=self.model_data_prefix,
+            container_mode=self.container_mode,
+        )
+
+    def add_model(self, s3_url, model_data_path=None):
+        """Adds a model to the `MultiDataModel` by copying
+        the s3_url model artifact to the given S3 path relative to model_data_prefix
+
+        Args:
+        s3_url: S3 path of the trained model artifact
+        model_data_path: S3 path where the trained model artifact
+                should be uploaded relative to `self.model_data_prefix`.
+                (default: None). If None, then the entire s3_url path
+                (after the source bucketname) will be copied to
+                `model_data_prefix` location
+        """
+        # Validate s3_url
+        if not s3_url.startswith("s3://"):
```

Comment:
Is it possible that we also support local files? Python SDK can take care of uploading local model files to S3 paths using S3Uploader: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/s3.py#L37

Reply:
Updated logic to support local files as well. Thanks for the suggestion